diff --git a/dist/index.js b/dist/index.js
index a09429e5f49282cbc2cb763ad653d63497b2baa6..af6fabe43b30a36f5a1c88ac78d3ff7192ba8540 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -34484,7 +34484,8 @@ function createServerHandler(config) {
       if (error instanceof QueryEngineHttpError) {
         res.status(error.statusCode).send(error.responseBody);
       } else {
-        res.status(500).json({ error }).end();
+        debug3("!!!error:", error);
+        res.status(500).end();
       }
     }
   });